Client/game_sa/CStreamingSA.cpp

```diff
@@ -133,4 +133,10 @@ void CStreamingSA::RequestSpecialModel(DWORD model, const char* szTexture, DWORD
         call    dwFunc
         add     esp, 0xC
     }
-}
\ No newline at end of file
+}
+
+CStreamingInfo* CStreamingSA::GetStreamingInfoFromModelId(ushort id)
+{
+    CStreamingInfo* pItemInfo = (CStreamingInfo*)(ARRAY_StreamModelInfo);
+    return &pItemInfo[id];
+}
```

Comment:
There's a nice way to do this: 
```
// in CStreamingSA.h
class CStreamingSA
{
    public:
    static CStreamingInfo(&s_aInfoForModel)[26316];
};

// in CStreamingSA.cpp
// no need to use ARRAY_StreamModelInfo preprocessor
CStreamingInfo(&CStreamingSA::ms_aInfoForModel)[26316] = *(CStreamingInfo(*)[26316])0x8E4CC0;
```

Then you can use it just like a normal array:
```
CStreamingInfo* CStreamingSA::GetStreamingInfoFromModelId(ushort id)
{
    return &ms_aInfoForModel[id];
}
```